### PR TITLE
py-scikit-learn: add v1.5.1

### PR DESCRIPTION
**Pull request by adamjstewart**
https://scikit-learn.org/stable/whats_new/v1.5.html#version-1-5-1

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-meson-python/package.py [hunk 1]
```diff
@@ -25,12 +26,17 @@ class PyMesonPython(PythonPackage):
...
+        "0.7.0",
+        sha256="9fcfa350f44ca80dd4f5f9c3d251725434acf9a07d9618f382e6cc4629dcbe84",
+        deprecated=True,
+    )
 
     depends_on("py-colorama", when="platform=windows", type=("build", "run"))
     depends_on("meson@0.63.3:", when="@0.11:", type=("build", "run"))
     depends_on("meson@0.63:", when="@0.9:0.10", type=("build", "run"))
     depends_on("meson@0.62:", type=("build", "run"))
+    depends_on("py-packaging@19:", when="@0.16:", type=("build", "run"))
```
Comment (rgommers) [inline var/spack/repos/builtin/packages/py-meson-python/package.py:39]: Why is `packaging` needed as a dependency? It's not in the upstream `pyproject.toml`.

Comment (adamjstewart) [reply var/spack/repos/builtin/packages/py-meson-python/package.py:39]: https://github.com/mesonbuild/meson-python/blob/0.16.0/pyproject.toml#L38

Comment (rgommers) [reply var/spack/repos/builtin/packages/py-meson-python/package.py:39]: Ah sorry, reading comprension problem on my side. I thought it was a `scikit-learn` update. Yes, for `meson-python` it is right.

**Review by rgommers — APPROVED** [review]
This all LGTM, thanks @adamjstewart. Looks like the CI failure is a timeout on `py-torch`, which is unrelated.

**Comment by adamjstewart** [comment]
@spackbot run pipeline

**Review by alalazo — APPROVED** [review]

**Pull request merged**